### PR TITLE
Improvements Discord Webhook

### PR DESCRIPTION
**Pull request by Daenges**
Improve caching and remove unused functions

### CODE REVIEW

**Comment by ORelio** [comment]
Oops, did not see you already opened a new PR so I kept posting in #1532. So, as I said, maybe you could get rid of the Mojang API part by registering players in the dictionary inside `OnPlayerJoin`/`OnPlayerLeave` event of the ChatBot API?

**Comment by Daenges** [comment]
I just have some concerns about overloading this Dict.
Lets say, you have a big server, where several hundred players join and leave each day. If you run this bot for several days, the dictionary could have several thousand entries. Do you think that I should implement a clear of this list after a certain time? Problem there is that players who got enlisted one second too early, could be deleted and a new mojang request must be started. My sollution for this would be a list of a new player class which can hold a date additionally to the UUID and the name. Then you could clear all entries older than 10 min. Do you think this is necessary? @ORelio

**Comment by ORelio** [comment]
Well, overestimating the size of one entry to 512 bytes and taking 1000 players * 10 days * 512, that's still only like ~5MB of RAM.

**Comment by Daenges** [comment]
But this will also implies a search through more and more data, just because one Player was not found, which could take some time.

**Comment by ORelio** [comment]
A dictionary is a data structure made to index the values by key and lookup the data immediately regardless of its size.

> Each addition to the dictionary consists of a value and its associated key. Retrieving a value by using its key is very fast, close to O(1), because the Dictionary<TKey,TValue> class is implemented as a hash table.

Source: https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2?view=net-5.0#remarks

> In a well-dimensioned hash table, the average cost (number of instructions) for each lookup is independent of the number of elements stored in the table.

Source: https://en.wikipedia.org/wiki/Hash_table

**Comment by Daenges** [comment]
I was actually a bit concerned because of the lookup of .ContainsKey(). But as described below, it is nothing else than a try-catch with the value.
Source: https://stackoverflow.com/questions/9382681/what-is-more-efficient-dictionary-trygetvalue-or-containskeyitem

Okay. So in your opinion I can leave everything as it is?

**Comment by ORelio** [comment]
Good catch, TryGetValue is indeed better because you do only one lookup instead of two.
Yes, just adding mappings in the dictionary in memory seems fine to me since it'll clear when closing the client.
You may also clear the dictionary on connection lost, as servers may restart from time to time you can use this to perform cleanup.

File: MinecraftClient/config/ChatBots/DiscordWebhook.cs [hunk 1]
```diff
@@ -101,6 +101,7 @@ public WebhoookSettings()
     public Dictionary<string, List<string>> GetCachedMessages() { return this.messageCache; }
     public Dictionary<string, string> GetSkinModes() { return this.skinModes; }
     public Dictionary<string, string> GetNamesToUuidMojangCache() { return this.namesToUuidMojangCache; }
+    public void resetUUIDCache() { namesToUuidMojangCache = new Dictionary<string, string>(); }
```
Comment (ORelio) [inline MinecraftClient/config/ChatBots/DiscordWebhook.cs:104]:
```suggestion
    public void resetUUIDCache() { namesToUuidMojangCache.Clear(); }
```

**Review by ORelio — APPROVED** [review]
If you are done with the changes let me know if I can merge the PR

**Comment by Daenges** [comment]
@ORelio Everything is done here. :)

**Pull request merged**